spec/abstract/IpcCancel_A.thy

```diff
@@ -239,6 +239,21 @@ where
                | _ \<Rightarrow> return True
   od"
 
+definition
+  bind_sc_reply :: "obj_ref \<Rightarrow> obj_ref \<Rightarrow> (unit, 'z::state_ext) s_monad"
+where
+  "bind_sc_reply sc_ptr reply_ptr  = do
+      sc_replies \<leftarrow> liftM sc_replies $ get_sched_context (sc_ptr);
+      \<comment> \<open>unlink head reply and sc before pushing\<close>
+      case sc_replies of
+          [] \<Rightarrow> return ()
+        | (r#_) \<Rightarrow> set_reply_obj_ref reply_sc_update r None;
```

Comment:
```suggestion
        | r#_ \<Rightarrow> set_reply_obj_ref reply_sc_update r None;
```
avoid unnecessary parentheses

Reply:
It feels like there should be a neater way to express this case on option, but I'd have to fire up Isabelle to find it.

Reply:
It would be like `maybeM` but for lists. Or otherwise using `list_option`.